### PR TITLE
Add short description and section for how to use cmcli to README.md

### PR DESCRIPTION
**Pull request by martin-ejdestig**
Also mention ConnMan in list of dependencies, but indicate that it is not meant to be a "hard" dependency.

When more extensive documentation is added in e.g. a doc/ directory, some information in README.md should probably be moved there (e.g. just mention that cmcli exists in README.md and link to relevant section in doc/).

### CODE REVIEW

**Comment by martin-ejdestig** [comment]
@JEderonn Not entirely happy with initial section. Perhaps you have any better ideas for what to put there?

**Comment by TSAnsari** [comment]
Perhaps a shorter commit message for eg,
README.md: add cmcli how-to section

**Comment by JEderonn** [comment]
> @JEderonn Not entirely happy with initial section. Perhaps you have any better ideas for what to put there?

I have already approved, but I wonder if it would make sense to write something where the D-Bus interface is located, i.e. https://github.com/Pelagicore/connectivity-manager/blob/master/data/com.luxoft.ConnectivityManager.xml and some words about the configuration, what is configured etc.

**Comment by JEderonn** [comment]
> README.md: add cmcli how-to section

rephrase to
README.md: add how-to section for cmcli
or
README.md: add how-to section for command-line interface cmcli

**Comment by martin-ejdestig** [comment]
> > README.md: add cmcli how-to section
> 
> rephrase to
> README.md: add how-to section for cmcli
> or
> README.md: add how-to section for command-line interface cmcli

Too short. I can split it out in two commits though. One "README.md: add short description" and "README.md: add section describing how to use cmcli" or something.

Never been fond of the "<foo>: bar" format. It leaves less space to write actual text. It is weird when there is more than a single file or component that is modified. And... Git knows what files you modify and it ("foo:") can perhaps be generated if you really want it. But if you want, I will start using it in this repo as well when there is enough space and a clear file/component. But as far as I know we do not have a requirement on using it. https://wiki.gnome.org/Git/CommitMessages says "you can add a tag", not "must".
</braindumpontagincommitmessages> :)

**Comment by martin-ejdestig** [comment]
> > @JEderonn Not entirely happy with initial section. Perhaps you have any better ideas for what to put there?
> 
> I have already approved, but I wonder if it would make sense to write something where the D-Bus interface is located, i.e. https://github.com/Pelagicore/connectivity-manager/blob/master/data/com.luxoft.ConnectivityManager.xml

Sure, I can add something about the "goal" with the D-Bus interface in the XML file as well.

(Note how if I do this in the same commit message, the tag: format of short message becomes... weird. ;)

> and some words about the configuration, what is configured etc.

Hum, do not understand what you mean here. "configuration"?

**Comment by JEderonn** [comment]
> Hum, do not understand what you mean here. "configuration"?

Well, I though about https://github.com/Pelagicore/connectivity-manager/blob/master/data/com.luxoft.ConnectivityManager.conf but perhaps not needed as it's a normal D-Bus configuration.

**Comment by JEderonn** [comment]
> Too short. I can split it out in two commits though. One "README.md: add short description" and "README.md: add section describing how to use cmcli" or something.

Sure, split into multiple commits or keep the commit header "as is". I would approve both

**Comment by TSAnsari** [comment]
> > > README.md: add cmcli how-to section
> > 
> > 
> > rephrase to
> > README.md: add how-to section for cmcli
> > or
> > README.md: add how-to section for command-line interface cmcli
> 
> Too short. I can split it out in two commits though. One "README.md: add short description" and "README.md: add section describing how to use cmcli" or something.
> 
> Never been fond of the "tag: foo bar" format. It leaves less space to write actual text. It is weird when there is more than a single file or component that is modified. And... Git knows what files you modify and it ("tag:") can perhaps be generated if you really want it. But if you want, I will start using it in this repo as well when there is enough space and a clear file/component. But as far as I know we do not have a requirement on using it. https://wiki.gnome.org/Git/CommitMessages says "you can add a tag", not "must".
> </braindumpontagincommitmessages> :)

It was just a suggestion. I don't have any strong opinions about "tag: foo bar" format. I use it when I feel that it will make the commit message compact/understandable. Also it helps in human readability when identify what component or file this commit is related to.

File: data/com.luxoft.ConnectivityManager.xml [hunk 1]
```diff
@@ -4,6 +4,11 @@
 <!-- TODO: Note that API should not be considered stable yet and is still
      subject to change. Should go through all TODOs before removing this note. -->
 <node>
+  <!--
+      Main interface for connectivity-manager.
+
+      Goal with API is to make it easy to implement graphical user interfaces.
```
Comment (martin-ejdestig) [inline data/com.luxoft.ConnectivityManager.xml]: If you have any ideas for how to reword/improve this, please come with suggestions. :)

**Comment by martin-ejdestig** [comment]
> I have already approved, but I wonder if it would make sense to write something where the D-Bus interface is located, i.e. https://github.com/Pelagicore/connectivity-manager/blob/master/data/com.luxoft.ConnectivityManager.xml 

Added a short description for the main interface mentioning what goal with API is. Could be improved though... but I am out of ideas and too snowed in in my own thinking at the moment, i think.

File: data/com.luxoft.ConnectivityManager.xml (hunk 1 above)
Comment (JEderonn) [reply data/com.luxoft.ConnectivityManager.xml]: Sure, not easy but here is a proposal :).
Used to control and retrieve information of all supported technologies, e.g. WiFi connectivity.

API goals are easy-of-use and provide unique functionality to be used in an automotive context.

Comment (martin-ejdestig) [reply data/com.luxoft.ConnectivityManager.xml]: @JEderonn Hum, reading your comment about adding something in the XML file again... I am not sure what I added is what you were after.

Comment (martin-ejdestig) [reply data/com.luxoft.ConnectivityManager.xml]: Sounds better.

**Comment by martin-ejdestig** [comment]
> He he, what I originally meant was that it might have been good with a link to where the D-Bus API is located. Sure, you can browse the code tree but a link to the file where it's located is convenient.

Ah, I see. Added a relative link.

**Pull request merged**